### PR TITLE
support windows in ci

### PR DESCRIPTION
**Pull request by aryairani**
This will initially fail CI but will be a starting point for getting a Windows build working to the point of passing CI.

### CODE REVIEW

File: .gitattributes
Comment (ChrisPenner) [inline .gitattributes:1]: This file just says to always normalize line-endings for git in this repo, even if a user's local settings say otherwise.

File: .gitattributes
Comment (ChrisPenner) [inline .gitattributes:2]: This is probably unnecessary, but it doesn't hurt to tell git about our custom binary filetype since it's not very common.

**Comment by ChrisPenner** [comment]
I've merged #2948  into here for testing while waiting for review; but after it gets reviewed and merged I'm hopeful we'll have a passing build and be able to merge 👍🏼 

This PR Includes:

* Change transcript diff detection to use `git diff` with flags to ignore carriage returns at EOL (since transcript outputs will have carriage returns on windows)
* Add a gitattributes file to ensure files committed on windows will have their line endings normalized

**Review by ChrisPenner — APPROVED** [review]

**Comment by aryairani** [comment]
This is great, thanks.  Let's squash and rebase IMO once #2948 is merged.

**Comment by aryairani** [comment]
Oh but also un-disable the other platforms' CI.

**Comment by ChrisPenner** [comment]
Needed to change how we install stack so it's not in the same working directory as the unison repo (so that the new git diff check works), but I think it's probably a good idea to keep the checkout dir clean anyways 😄 . Mergify has shunned me since the ci file has changed, but this is ready to merge (if/when) CI passes 👍🏼

File: .github/workflows/ci.yaml
Comment (aryairani) [inline .github/workflows/ci.yaml:27]: We should drop this line ^

**Pull request merged**
